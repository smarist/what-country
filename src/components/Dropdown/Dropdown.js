import {
  ButtonBase,
  FormControl, InputLabel, MenuItem, Select,
} from '@material-ui/core';
import React from 'react';
import useStyles from './Dropdown.style';

function Dropdown({
  label,
  onSelect,
  value,
  options,
  variant,
  styles,
}) {
  const classes = useStyles();
  return (
    <FormControl
      fullWidth
      variant={variant || 'outlined'}
      className={classes.formControl}
      style={styles}
    >
      {label && <InputLabel id="demo-customized-select-label">{label}</InputLabel> }
      <Select
        labelId="demo-customized-select-label"
        id="demo-customized-select"
        value={value}
        onChange={e => {
          onSelect(e.target.value);
        }}
        // input={<BootstrapInput />}
      >
        {options.map((option, index) => (
          <MenuItem
            selected={index === 0 ? true : undefined}
            className={classes.optionsMenuItem}
            value={option.value}
            key={`menu-item${index}`}
          >
            <ButtonBase
              disableRipple
              className={classes.rippleButton}
              style={{
                ...option.itemStyle,
              }}
              onClick={() => option.onClick?.(option)}
            >
              {option.name}
            </ButtonBase>
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
}

export default Dropdown;
