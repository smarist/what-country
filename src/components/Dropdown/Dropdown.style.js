import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  margin: {
    margin: theme.spacing(1),
  },
  formControl: {
    '& fieldset': {
      border: 'none',
    },
    '& label': {
      color: theme.palette.placeHolderColor.main,
      fontSize: '18px',
      marginTop: 0,
      width: '80%',
      display: 'block',
      whiteSpace: 'nowrap',
      textOverflow: 'ellipsis',
      overflow: 'hidden',
    },
    '& label.MuiInputLabel-shrink': {
      color: theme.palette.placeHolderColor.main,
      fontSize: '18px',
      marginTop: -8,
    },
    '& .MuiButtonBase-root': {
      color: theme.palette.textColor.main,
    },
  },
  rippleButton: {
    color: '#000',
  },
  input: {
    borderRadius: '3px',
    height: '35px',
    width: '100%',
    color: '#000',
    boxShadow: 'none',
    fontSize: '14px',
    paddingLeft: 2,
    paddingTop: 5,
    border: 'none',
    overflow: 'hidden',
    '& > div:first-child': {
      zIndex: 5,
      width: '100%',
      fontWeight: 400,
      paddingLeft: 5,
    },
    '&.Mui-focused > div:first-child': {
      backgroundColor: 'transparent',
    },
    '&.avatar-wrapper': {
      color: '#011B60',
    },
    '&.avatar-wrapper img': {
      backgroundColor: theme.palette.textColor.main,
      color: theme.palette.textColor.main,
      width: '100%',
      height: '100%',
      paddingTop: 12,
      paddingBottom: 12,
      justifyContent: 'flex-start',
    },
    '& path': {
      stroke: theme.palette.textColor.main,
    },
  },
  label: {
    marginTop: -7,
    fontSize: 16,
  },
  dropDownValue: {
    fontSize: '13px',
    fontWeight: 400,
    padding: '12px 15px',
    color: '#011B60',
  },
}));

export default useStyles;
