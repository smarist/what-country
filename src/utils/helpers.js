export function objectToArray(val) {
  const obj = val;
  return Object.entries(obj);
}

let timeoutId;

export function debounceHandler(func, delay) {
  if (timeoutId) clearTimeout(timeoutId);
  timeoutId = setTimeout(func, delay);
}
