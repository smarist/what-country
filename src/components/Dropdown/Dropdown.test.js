import { render } from '@testing-library/react';
import { withThemeProvider } from '../../tests/helpers';
import Dropdown from './Dropdown';

describe('Dropdown Test', () => {
  it('should render successfully', () => {
    const { queryByTestId } = render(withThemeProvider(Dropdown, { value: '' }));
    const dropdown = queryByTestId('dropdown');

    expect(dropdown).toBeTruthy();
    expect(dropdown).toMatchSnapshot();
  });
});
