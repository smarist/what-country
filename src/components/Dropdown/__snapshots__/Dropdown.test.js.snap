// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Test should render successfully 1`] = `
<div
  class="MuiFormControl-root makeStyles-formControl-2 MuiFormControl-fullWidth"
  data-testid="dropdown"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root  makeStyles-input-4 MuiInputBase-formControl"
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby="demo-customized-select-label demo-customized-select"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      id="demo-customized-select"
      role="button"
      tabindex="0"
    >
      <span>
        ​
      </span>
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput"
      tabindex="-1"
      value=""
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend-8"
        style="width: 0.01px;"
      >
        <span>
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
